plugins/slicer/MONAILabel/MONAILabelLib/client.py

```diff
@@ -168,7 +169,8 @@ def download_label(self, label_id, tag):
                 MONAILabelError.SERVER_ERROR, f"Status: {status}; Response: {response}", status, response
             )
 
-        local_filename = tempfile.NamedTemporaryFile(dir=self._tmpdir, suffix=".nii.gz").name
+        file_ext = slicer.util.settingsValue("MONAILabel/fileExtension", ".nii.gz")
```

Comment:
i would rather not bring slicer import here.. let it be pure client api related functions (which later can be kind of common library for all py clients)

instead you can take the argument file_ext for `def download_label(self, label_id, tag)` and it can be passed from slicer related class/lib

another thing is, as @lassoan mentioned, this is for download.. which means the file extension is already correctly sent in `filename` from the server. 

https://github.com/Project-MONAI/MONAILabel/blob/main/monailabel/endpoints/datastore.py#L138


you can use the same for saving.. in that case the logic can be something like..
```
def download_label(self, label_id, tag, file_ext=".nii.gz")
  file_ext = "".join(pathlib.Path(file.filename).suffixes) if file.filename else file_ext
```